lib/matplotlib/quiver.py

```diff
@@ -902,23 +905,26 @@ class Barbs(mcollections.PolyCollection):
     # 1 triangle and a series of lines.  It works fine as far as I can tell
     # however.
     @docstring.interpd
-    def __init__(self, ax, *args, **kw):
+    def __init__(self, ax, *args,
+                 pivot='tip', length=7, barbcolor=None, flagcolor=None,
+                 sizes=None, fill_empty=False, barb_increments=None,
+                 rounding=True, flip_barb=False, **kw):
         """
         The constructor takes one required argument, an Axes
         instance, followed by the args and kwargs described
         by the following pylab interface documentation:
         %(barbs_doc)s
         """
-        self._pivot = kw.pop('pivot', 'tip')
-        self._length = kw.pop('length', 7)
-        barbcolor = kw.pop('barbcolor', None)
-        flagcolor = kw.pop('flagcolor', None)
-        self.sizes = kw.pop('sizes', dict())
-        self.fill_empty = kw.pop('fill_empty', False)
-        self.barb_increments = kw.pop('barb_increments', dict())
-        self.rounding = kw.pop('rounding', True)
-        self.flip = kw.pop('flip_barb', False)
+        self.sizes = sizes or dict()
+        self.fill_empty = fill_empty
+        self.barb_increments = barb_increments or dict()
+        self.rounding = rounding
+        self.flip = flip_barb
         transform = kw.pop('transform', ax.transData)
```

Comment:
Is there a reason not to convert *transform*?

Reply:
If `None` could be a user-supplied value, yes.  Otherwise no!

For this pull I'm really trying to make a small, obviously correct, incremental step.  I can't clean it up all at once, and I'm not going to try - I'd go crazy before it was debugged, let alone reviewed...

Reply:
I see the value of small incremental steps and agree with the strategy.

Indeed, explicitly providing ``transform=None`` is currently accepted and results in an ``IdentityTransform``. I suppose this is not intended, but should be discussed as a separate issue.

Reply:
I think None as IdentityTransform is fine...

Reply:
@anntzer I don't think so, because if not given, the default is ax.transData and in a quick test, I didn't get any reasonable output when using ``quiver(..., transform=None)`` explicitly. But let's not go there here. I'll open an issue for this soon when I have time to write it up correctly.

Reply:
See #11153

Reply:
... all artists have a default of transform=None maps to IdentityTransform.  Ummm, why?  I have no idea.  Back in the day, I think more was done in pixel co-ordinates.